/**
 * Copyright 2019 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package mindspore.dataset;

enum DataTypePb {
  DE_PB_UNKNOWN = 0;
  DE_PB_BOOL = 1;
  DE_PB_INT8 = 2;
  DE_PB_UINT8 = 3;
  DE_PB_INT16 = 4;
  DE_PB_UINT16 = 5;
  DE_PB_INT32 = 6;
  DE_PB_UINT32 = 7;
  DE_PB_INT64 = 8;
  DE_PB_UINT64 = 9;
  DE_PB_FLOAT16 = 10;
  DE_PB_FLOAT32 = 11;
  DE_PB_FLOAT64 = 12;
  DE_PB_STRING = 13;
}

message TensorPb {
  repeated int64 dims = 1; // tensor shape info
  DataTypePb tensor_type = 2; // tensor content data type
  bytes data = 3; // tensor data
}
